### PR TITLE
Add plugin-update-check workflow

### PR DESCRIPTION
**Pull request by kpcraig**
This PR is for hackweek (idea by @fairclothjm), adding a workflow that can be triggered from our plugin repos to create a PR specifically to test 'what would happen' if a plugin updated its dependencies.

The precipitating incident is a conflicting dependency update in vault-plugin-database-snowflake that had to be backed out due to a dependency conflict between it and vault.

The idea here is to have our plugin dependency workflow be able to call this workflow using 'repository_dispatch' and then ideally to be able to reference the PR thus generated back in the plugin PR.

Note: It is a bit unfortunate that we need this to be merged to `main`, but Github requires `repository_dispatch` actions to be on the default branch of a repository.

### CODE REVIEW

File: .github/workflows/plugin-update-check.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        run: |
+          echo "count=$(git status --porcelain=v1 2>/dev/null | wc -l)" >> "$GITHUB_OUTPUT"
+      - name: commit/push
+        if: steps.changes.outputs.count > 0
+        run: |
+          git config user.name hc-github-team-secure-vault-ecosystem
+          git config user.email hc-github-team-secure-vault-ecosystem@users.noreply.github.com
+          git add .
+          git commit -m "Automated dependency upgrades"
+          git push -f origin ${{ github.ref_name }}:"$BRANCH_NAME"
```
Comment (ncabatoff) [inline .github/workflows/plugin-update-check.yml:42]: I don't see where $BRANCH_NAME gets created?

Comment (kpcraig) [reply .github/workflows/plugin-update-check.yml:42]: Branch name is set in the env block up in line 20

**Review by fairclothjm — APPROVED** [review]

File: .github/workflows/plugin-update-check.yml (hunk 1 above)
Comment (ncabatoff) [reply .github/workflows/plugin-update-check.yml:50]: I saw the name being populated, but where is the branch itself being created?  Don't you need a `git checkout -b`?

Comment (ncabatoff) [reply .github/workflows/plugin-update-check.yml:50]: Oh, I see, this is a git push invocation I wasn't familiar with.

**Pull request merged**